core/app/views/refinery/sitemap/index.xml.builder

```diff
@@ -8,10 +8,16 @@ xml.urlset "xmlns" => "http://www.sitemaps.org/schemas/sitemap/0.9" do
      # exclude sites that are external to our own domain.
      page_url = if page.url.is_a?(Hash)
        # This is how most pages work without being overriden by link_url
-       page.url.merge({:only_path => false})
+       page.url.merge({:only_path => false, locale: locale})
      elsif page.url.to_s !~ /^http/
        # handle relative link_url addresses.
-       [request.protocol, request.host_with_port, page.url].join
+       raw_url = [request.protocol, request.host_with_port, page.url].join
+       if (@locales.size > 1) && defined?(RoutingFilter::RefineryLocales)
+         filter = RoutingFilter::RefineryLocales.new
+         filter.around_generate({}) do
+           raw_url
+         end
+       end
```

Comment:
You are right! Need to add else case to return raw_url without changes, or do some refactoring to get more effective correct code.